process/coordinator/process.go

```diff
@@ -1539,13 +1665,36 @@ func (tc *transactionCoordinator) verifyFees(
 }
 
 func (tc *transactionCoordinator) getMaxAccumulatedAndDeveloperFees(
+	miniBlockHeaderHandler data.MiniBlockHeaderHandler,
 	miniBlock *block.MiniBlock,
 	mapHashTx map[string]data.TransactionHandler,
+	processedMiniBlocks *processedMb.ProcessedMiniBlockTracker,
 ) (*big.Int, *big.Int, error) {
 	maxAccumulatedFeesFromMiniBlock := big.NewInt(0)
 	maxDeveloperFeesFromMiniBlock := big.NewInt(0)
 
-	for _, txHash := range miniBlock.TxHashes {
+	miniBlockHash, err := core.CalculateHash(tc.marshalizer, tc.hasher, miniBlock)
+	if err != nil {
+		return big.NewInt(0), big.NewInt(0), err
+	}
+
+	indexOfLastTxProcessedByItself := int32(-1)
+	if processedMiniBlocks != nil {
+		processedMiniBlockInfo, _ := processedMiniBlocks.GetProcessedMiniBlockInfo(miniBlockHash)
+		indexOfLastTxProcessedByItself = processedMiniBlockInfo.IndexOfLastTxProcessed
+	}
+
+	indexOfLastTxProcessedByProposer := miniBlockHeaderHandler.GetIndexOfLastTxProcessed()
+
+	for index, txHash := range miniBlock.TxHashes {
+		if index <= int(indexOfLastTxProcessedByItself) {
```

Comment:
please remove the conditionals and adapt instead the for init and stop values

Reply:
Done